database-commons/src/main/java/io/cdap/plugin/db/batch/action/AbstractDBArgumentSetter.java

```diff
@@ -0,0 +1,141 @@
+    Class<? extends Driver> driverClass = context.loadPluginClass(JDBC_PLUGIN_ID);
+    FailureCollector failureCollector = context.getFailureCollector();
+    SettableArguments settableArguments = context.getArguments();
+    processArguments(driverClass, failureCollector, settableArguments);
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer)
+      throws IllegalArgumentException {
+    DBUtils.validateJDBCPluginPipeline(pipelineConfigurer, config, JDBC_PLUGIN_ID);
+    Class<? extends Driver> driverClass = DBUtils.getDriverClass(
+        pipelineConfigurer, config, ConnectionConfig.JDBC_PLUGIN_TYPE);
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    FailureCollector collector = stageConfigurer.getFailureCollector();
+    config.validate(collector);
+    try {
+      processArguments(driverClass, collector, null);
+    } catch (SQLException e) {
+      collector.addFailure("SQL error while executing query: " + e.getMessage(), null)
+          .withStacktrace(e.getStackTrace());
+    } catch (IllegalAccessException | InstantiationException e) {
+      collector.addFailure("Unable to instantiate JDBC driver: " + e.getMessage(), null)
+          .withStacktrace(e.getStackTrace());
+    } catch (Exception e) {
+      collector.addFailure(e.getMessage(), null).withStacktrace(e.getStackTrace());
+    }
+    config.validate(collector);
+  }
+
+  /**
+   * Creates connection to database. Reads row from database based on selection conditions and
+   * depending on whether settable arguments is provided or not set the argument from row columns.
+   *
+   * @param driverClass {@link Class<? extends Driver>}
+   * @param failureCollector {@link FailureCollector}
+   * @param settableArguments {@link SettableArguments}
+   * @throws SQLException is raised when there is sql related exception
+   * @throws IllegalAccessException is raised when there is access related exception
+   * @throws InstantiationException is raised when there is class/driver issue
+   */
+  private void processArguments(Class<? extends Driver> driverClass,
+      FailureCollector failureCollector, SettableArguments settableArguments)
+      throws SQLException, IllegalAccessException, InstantiationException {
+    DriverCleanup driverCleanup;
+
+    driverCleanup = DBUtils.ensureJDBCDriverIsAvailable(driverClass, config.getConnectionString(),
+        config.jdbcPluginName);
+    Properties connectionProperties = new Properties();
+    connectionProperties.putAll(config.getConnectionArguments());
+    try {
+      Connection connection = DriverManager
+          .getConnection(config.getConnectionString(), connectionProperties);
+      Statement statement = connection.createStatement();
+      ResultSet resultSet = statement.executeQuery(config.getQuery());
+      boolean hasRecord = resultSet.next();
+      if (!hasRecord) {
+        failureCollector.addFailure("Ne record found",
```

Comment:
Typo: `No records found.`
Please add period to every failure message and corresponding corrective action

Reply:
@CuriousVini updated messages to have corrective action.